src/main/java/io/vavr/control/Either.java

```diff
@@ -483,6 +483,50 @@ public final void orElseRun(Consumer<? super L> action) {
         }
     }
 
+    /**
+     * Calls recoverFunction if the projected Either is a Left, performs no operation if this is a Right. This is
+     * similar to {@code getOrElseGet} where the fallback method also returns an Either.
+     *
+     * <pre>{@code
+     * Either<Integer, String> tryGetString() { return Either.left(1); }
+     *
+     * Either<Integer, String> tryGetStringAnotherWay(Integer lvalue) { return Either.right("yo " + lvalue); }
+     *
+     * = Right("yo 1")
+     * tryGetString().recover(this::tryGetStringAnotherWay);
+     * }</pre>
+     *
+     * @param recoverFunction a function which accepts a Left value and returns an Either
+     * @return an {@code Either<L, R>} instance
+     */
+    public final Either<L, R> recover(Function<L, Either<L, R>> recoverFunction) {
+        if (isLeft()) {
+            return recoverFunction.apply(getLeft());
+        } else {
+            return this;
+        }
+    }
+
+    /**
+     * Calls recoverSupplier if the projected Either is a Left, performs no operation if this is a Right. This is
+     * similar to {@code getOrElseGet} where the fallback method also returns an Either.
+     *
+     * <pre>{@code
+     * Either<Integer, String> tryGetString() { return Either.left(1); }
+     *
+     * Either<Integer, String> tryGetStringAnotherWay() { return Either.right("yo"); }
+     *
+     * = Right("yo")
+     * tryGetString().recover(this::tryGetStringAnotherWay);
+     * }</pre>
+     *
+     * @param recoverSupplier a {@code Supplier} which returns an Either
+     * @return an {@code Either<L, R>} instance
+     */
+    public final Either<L, R> recover(Supplier<Either<L, R>> recoverSupplier) {
```

Comment:
Currently I simplify VAVR by reducing the API surface area. Instead of adding an additional recover / recoverWith function taking a Supplier, we stick to a recover `Function` argument only.

Here is how it looks in `Try` for VAVR 2.0:

https://github.com/vavr-io/vavr/blob/bec7385433eb5e4273e2eb71cc86acc93191000f/src/main/java/io/vavr/control/Try.java#L551-L630

Could you therefore please add instead this function:

```java
Either<L, R> recover(Function<? super L, ? extends R> recoverFunction)
```

Thanks!

Reply:
One more thing regarding the wording:

* in Try the parameter is named `recoveryFunction`
* in Either you named it `recoverFunction`

I'm no native speaker. Which one is better? We should sync both.

Reply:
I think `recoveryFunction` is better actually, so I've renamed appropriately.

I'm not sure I understand your Try example - I think a method `Either<L, R> recover(Function<? super L, ? extends R> recoverFunction)` would be functionally equivalent to just calling `getOrElseGet`. The purpose of the `Supplier` overload is for convenience if you don't care what the L-value was when trying to recover (whereas the `Function` overload receives it as a parameter, potentially allowing you to do different things based on what error occurred).

However, I understand your point about simplifying the API, so I'm happy to just remove it completely if you prefer.

Reply:
Thx! It still makes sense to have `recover(Function)` because I will align `getOrElse` and `getOrElseGet` to Scala in VAVR 2.0. In VAVR 1.0 we will have `getOrElseGet(Function)`. Starting with VAVR 2.0, it will be `getOrElseGet(Supplier)`. **So, please implement `recover(Function)`** :)

In Scala, Try looks like this:

<img width="675" alt="Screenshot 2019-08-21 at 13 58 39" src="https://user-images.githubusercontent.com/743833/63430149-2b886c00-c41c-11e9-9fe4-b280bca5015a.png">

<img width="734" alt="Screenshot 2019-08-21 at 13 58 55" src="https://user-images.githubusercontent.com/743833/63430159-3216e380-c41c-11e9-9a36-09ce4fc14ffe.png">

The `=> U` type indicates in Scala, that it is lazily evaluated. We do not have such a feature in Java, so we will have `getOrElseGet(Supplier)` in VAVR 2.0.

The partial recovery function in Scala is a hint that pattern matching is possible in that location. I don't think that Java's native pattern matching will implement j.u.f.Function. We will see what we can do by time...

Reply:
OK, I've implemented what I think you are expecting. Let me know if it's ok